core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -305,7 +325,7 @@ class ReplicaManager(val config: KafkaConfig,
                     responseCallback: Map[TopicPartition, PartitionResponse] => Unit) {
 
     if (isValidRequiredAcks(requiredAcks)) {
-      val sTime = time.milliseconds
+      val sTime = if (logger.isDebugEnabled) time.milliseconds else 0
```

Comment:
By the way, I think the code as it is now is a bit dangerous. Imagine that there are some changes and this field starts being used in info logging. It's not unlikely that a bug will be introduced. Is this optimisation really worth doing?